conda/common/logic.py

```diff
@@ -313,7 +318,7 @@ def Assign_(self, vals, name=None):
         x = self._assign_no_name(vals)
         if not name:
             return x
-        if isinstance(x, bool):
+        if vals in {TRUE, FALSE}:
```

Comment:
Note that the `x` -> `vals` change is correct here. That was due to an oversight in https://github.com/mbargull/conda/commit/e1d34c277906ff1512f766ab8806d4040a70c184 that changed `vals` -> `x`.
(Either way, they are functionally identical since `x` is a Boolean if and only if `vals` is a Boolean here.)